### ISSUE TITLE
(SQL server) Collation when creating a database

### ISSUE THREAD

**Issue by MihailsKuzmins** [Zaid-Ajaj/ThrowawayDb#11]
Hi @Zaid-Ajaj,

When creating a new ThrowawayDatabase I need to specify a custom collation for the entire database, so my required SQL script is as follows:
```sql
CREATE DATABASE [MyDatabase] COLLATE MyCollation
```
Currently a workaround is possible by running this script after the ThrowawayDatabase  has been created, but wouldn't it be nicer if the collation could be specified already when creating the database?
```sql
ALTER DATABASE [MyDatabase] COLLATE MyCollation
```

If you think it's a good feature to have I could help with a PR. I have already considered an option how it can be done. My first idea was adding a new optional string parameter, but unfortunately it results in the overload problem of the `Create` method.
![image](https://user-images.githubusercontent.com/47413092/102617450-e07e8600-4141-11eb-86c9-ce9f8926cc5d.png)

Therefore, I think the only option would be adding a new class (e.g. `ThrowawayDatabaseOptions`) with all the required parameters (currently DatabaseNamePrefix and Collation), but it's a breaking change of the API.

**Comment by Zaid-Ajaj** [comment]
Hi @MihailsKuzmins, the option sounds like a good idea. Why do you think it becomes a breaking change if it just another overload for the `Create` static function that only takes `ThrowawayDatabaseOptions`?

**Comment by MihailsKuzmins** [comment]
Hi @Zaid-Ajaj,
huh, I guess you are right. I thought about replacing the current methods with the new class. But your your approach is indeed better, just create another overload and that's it 👍 
In this case I will proceed with how you suggested. Most probably not today (well it's Friday night :boom:), but tomorrow I could have some results

**Comment by Zaid-Ajaj** [comment]
That would be awesome @MihailsKuzmins ❤️ 🙏 I will be happy to merge and publish as soon you are done! No rush though 😉

**Comment by MihailsKuzmins** [comment]
Hi @Zaid-Ajaj,
speaking of tests. Currently it's just a console library which runs all methods. Don't you consider an opportunity to use an xUnit project instead?
I believe you use a VS Code, so there is a [.NET Core Test Explorer extension](https://marketplace.visualstudio.com/items?itemName=formulahendry.dotnet-test-explorer) for running unit tests in VS Code.

**Comment by Zaid-Ajaj** [comment]
I don't mind adding proper unit tests, I just couldn't find the time to actually do it 😅

**Comment by MihailsKuzmins** [comment]
> I don't mind adding proper unit tests, I just couldn't find the time to actually do it 😅

I will do it then :sunglasses:

**Comment by Zaid-Ajaj** [comment]
Thanks a lot @MihailsKuzmins really appreciate it!

**Comment by MihailsKuzmins** [comment]
Created a PR #12.
In order to get rid of the code duplication I needed to rewrite how static factory functions are called. Therefore, I added a couple of unit tests, so I believe I have covered all overloads with all possible input parameters :blush:. You could also run them on your PC to make sure that it works for you as well, not only me :sweat_smile:

**Comment by Zaid-Ajaj** [comment]
I've merged the PR to try out the tests and publish but now I can no longer see the PR page and I cannot pull latest code 😅 never had this before, can you view the PR page at #12?

**Comment by Zaid-Ajaj** [comment]
OK, never mind hahah the page has recovered. It was returning 500 for some reason :joy:

**Comment by Zaid-Ajaj** [comment]
Merged and publish #12 as of ThrowawayDb v1.4 🚀 

I had some issues when connecting to the database using a username/password combo but that seemed to be an issue on my machine only because even when I try logging in using SQL authentication for a login credentials that I have just added, it still doesn't work. So I just assumed the tests worked locally on your machine instead and pushed a version anyways 😉

**Comment by MihailsKuzmins** [comment]
> Merged and publish #12 as of ThrowawayDb v1.4 🚀
> 
> I had some issues when connecting to the database using a username/password combo but that seemed to be an issue on my machine only because even when I try logging in using SQL authentication for a login credentials that I have just added, it still doesn't work. So I just assumed the tests worked locally on your machine instead and pushed a version anyways 😉

Thank you🙏
Speaking of the authentication issue. Have you configured this setting of your SQL server instance? (Right clink on your server -> Properties -> Security). Otherwise if it's an *sa* user (or another user who is added to a database) authentication should work.
![image](https://user-images.githubusercontent.com/47413092/102697862-e3f93680-4241-11eb-9aca-8187804c128b.png)

**Comment by Zaid-Ajaj** [comment]
That worked!! 😍 Just ran all tests and all 31 of them passed ✔️
